taskana-adapter-test/src/test/java/pro/taskana/adapter/integration/TestTaskAcquisition.java

```diff
@@ -83,40 +89,73 @@ public void multiple_user_task_process_instances_started_in_camunda_via_rest_sho
+                .startCamundaProcessAndReturnId("simple_user_task_with_complex_variables_process", "");
+        List<String> camundaTaskIds = this.camundaProcessengineRequester
+                .getTaskIdsFromProcessInstanceId(processInstanceId);
 
         Thread.sleep(this.adapterTaskPollingInterval);
 
-        for (String camundaTaskId : camundaTaskIds) {
-            List<TaskSummary> taskanaTasks = this.taskService.createTaskQuery().externalIdIn(camundaTaskId).list();
-            assertEquals(1, taskanaTasks.size());
-            TaskSummary taskanaTaskSummary = taskanaTasks.get(0);
-            String taskanaTaskExternalId = taskanaTaskSummary.getExternalId();
-            assertEquals(taskanaTaskExternalId, camundaTaskId);
+        String assumedVariablesString = "{\"attribute1\":{\"type\":\"ProcessVariableTestObject\",\"value\":{\"stringField\":\"stringValue\",\"intField\":1,\"doubleField\":1.1,\"booleanField\":false,\"processVariableTestObjectTwoField\":{\"intFieldObjectTwo\":2,\"doubleFieldObjectTwo\":2.2,\"booleanFieldObjectTwo\":true,\"dateFieldObjectTwo\":\"1970-01-01 13:12:11\",\"stringFieldObjectTwo\":\"stringValueObjectTwo\"}},\"valueInfo\":{\"objectTypeName\":\"pro.taskana.impl.ProcessVariableTestObject\"}},\"attribute2\":{\"type\":\"String\",\"value\":\"attribute2Value\",\"valueInfo\":{\"objectTypeName\":\"java.lang.String\"}},\"attribute3\":{\"type\":\"String\",\"value\":\"attribute3Value\",\"valueInfo\":{\"objectTypeName\":\"java.lang.String\"}}}";
+        camundaTaskIds.forEach(camundaTaskId -> createTaskanaTaskAndVerifyTaskVariables(camundaTaskId,assumedVariablesString));
+    }
 
-            // get the actual task instead of summary to access custom attributes
-            Task taskanaTask = this.taskService.getTask(taskanaTaskSummary.getTaskId());
-            Map<String, String> taskanaTaskCustomAttributes = taskanaTask.getCustomAttributes();
-            String variablesKeyString = "referenced_task_variables";
-            String taskanaVariablesString = taskanaTaskCustomAttributes.get(variablesKeyString);
-            String assumedVariablesString = "{\"amount\":{\"type\":\"Long\",\"value\":555,\"valueInfo\":{}},\"item\":{\"type\":\"String\",\"value\":\"item-xyz\",\"valueInfo\":{}}}";
 
-            // attention: the order of "amount" and "item" returned by camunda is random. Therefore, a simple
-            // comparison of the strings doesn't work.
-            // rather use SameJSONAs.sameJSONAs from hamcrest-json to compare Json strings independent of child order
-            Assert.assertThat(
-                assumedVariablesString,
-                SameJSONAs.sameJSONAs(taskanaVariablesString));
-        }
+    @WithAccessId(
+            userName = "teamlead_1",
+            groupNames = {"admin"})
+    @Test
+    public void task_with_complex_variables_from_parent_execution_should_result_in_taskanaTasks_with_those_variables_in_custom_attributes()
```

Comment:
Please add a testcase where a variable contains a String that contains characters that must be escaped in Json like
 
\b  Backspace (ascii code 08)
\f  Form feed (ascii code 0C)
\n  New line
\r  Carriage return
\t  Tab
\"  Double quote
\\  Backslash character
Just to be sure that we transfer these variables correctly to taskana ...

Reply:
Done